README.md

```diff
@@ -1,14 +1,59 @@
+
+## High level breakdown
+
+### Online
+
+1. Tidal Tools
+    - Exposes the source code analysis command `tidal analyze code --app-id <your-app-id>` (run from directory containing the source code)
+    - Handles authentication
+    - Hits your Tidal Migrations Workspace `GET apps/#id/cast`
+2. Tidal Migrations Platform
+    - Creates an app in CAST if it's not already present. This CAST app is now associated with your TMP app at the ID provided.
+    - Returns the CAST ID of the app which was created OR the CAST ID of the existing CAST app if you're analyzing the same app again (premium users only).
+3. Tidal Tools
+    - Pulls docker image containing CAST Highlight CLI (that's why you're here)
+    - Runs the CAST CLI in this container
+4. CAST CLI
+    - Analyzes your app and uploads it to CAST with the CAST ID which TMP returned in step 2
+5. Tidal Tools
+    - After the analysis is completed successfully, Tidal Tools will notify TMP that the app has been analyzed
+6. Tidal Migrations Platform
+    - Starts a worker to check on the status of the analysis in CAST. This runs async and periodically checks to see if the analysis is complete
+    - When the analysis is complete, TMP gets the analysis data from CAST and updates your TMP app with the analysis information. This propagates to the frontend and the user is now able to see their source code analysis result when they view their app
+
+![image](https://user-images.githubusercontent.com/43866616/167871482-c18c3fc3-86dd-49d2-a0e9-178a92a7757c.png)
+
+### Offline
+
+1. Tidal Tools
+    - Exposes the source code analysis command `tidal analyze code --app-id <your-app-id>` (run from directory containing the source code). Providing the `--app-id` here is optional. If provided, the ID is saved as a comment with the ZIP file.
+    - Runs the CAST CLI in a docker container (obtained from running `tidal backup` on an offline server then transferring this over, see [the guide](https://guides.tidalmg.com/tidal-offline.html#create-the-tidal-tools-archive-file-for-offline-use).).
+    - Saves the resulting ZIP file to the current directory, or to a path provided.
+2. The User
+    - Sends this result ZIP file to Tidal Support.
+3. Tidal Support
+    - Uploads the result ZIP file to CAST on behalf of the user.
+    - Manually triggers the association between the Tidal Migrations Platform app and the CAST app.
+
+![image](https://user-images.githubusercontent.com/43866616/167871802-cb4f4ec6-375c-4a54-acf2-5602c9409222.png)
+
+## Requirements
+- Docker 
+- Gitlab
+    - Access to Tidal Tools repository [here](https://gitlab.com/subdata/tidal-tools)
+- GCP
+    - Images are stored in the [Container Registry](https://console.cloud.google.com/gcr/images/tidal-1529434400027/global/cast-highlight?authuser=1&project=tidal-1529434400027). You will need access to `Tidal Migrations` organization and to the `Tidal-Tools` project (id -> **tidal-1529434400027**)
```

Comment:
Removing this in case someone has multiple Gmail IDs logged in and Tidal's is not the first one.
```suggestion
    - Images are stored in the [Container Registry](https://console.cloud.google.com/gcr/images/tidal-1529434400027/global/cast-highlight?project=tidal-1529434400027). You will need access to `Tidal Migrations` organization and to the `Tidal-Tools` project (id -> **tidal-1529434400027**)
```

Reply:
Good catch!